packages/2d/src/components/SVG.ts

```diff
@@ -0,0 +1,684 @@
+ */
+export interface SVGDocument {
+  size: Vector2;
+  nodes: SVGShape[];
+}
+
+/**
+ * Data of SVGDocument.
+ * This can used many times because it do not reference parent SVG.
+ * This must build into SVGDocument
+ */
+export interface SVGDocumentData {
+  size: Vector2;
+  nodes: SVGShapeData[];
+}
+
+export interface SVGProps extends ShapeProps {
+  svg?: SignalValue<string>;
+}
+
+/**
+ * Node to render and animating SVG.
+ * You should use Image node if you do not want to animate children.
+ */
+export class SVG extends Shape {
+  @lazy(() => {
+    const element = document.createElement('div');
+    View2D.shadowRoot.appendChild(element);
+    return element;
+  })
+  protected static containerElement: HTMLDivElement;
+  private static svgNodesPool: Record<string, SVGDocumentData> = {};
+
+  /**
+   * SVG string to be rendered
+   */
+  @signal()
+  public declare readonly svg: SimpleSignal<string, this>;
+
+  /**
+   * Child to wrap all SVG node
+   */
+  public wrapper: Node;
+
+  private lastTweenTargetSrc: string | null = null;
+  private lastTweenTargetDocument: SVGDocument | null = null;
+
+  public constructor(props: SVGProps) {
+    super(props);
+    this.wrapper = new Node({});
+    this.wrapper.children(this.documentNodes);
+    this.add(this.wrapper);
+  }
+
+  /**
+   * Get SVG node by id.
+   * @param id - id
+   * @returns list of Node that have matched id
+   */
+  public getChildrenById(id: string) {
+    return this.document()
+      .nodes.filter(node => node.id === id)
+      .map(({shape}) => shape);
+  }
+
+  protected override desiredSize(): SerializedVector2<DesiredLength> {
+    const custom = super.desiredSize();
+    const {x, y} = this.document().size.mul(this.wrapper.scale());
+    return {
+      x: custom.x ?? x,
+      y: custom.y ?? y,
+    };
+  }
+
+  /**
+   * Convert SVGDocumentData into SVGDocument
+   * @param data - SVGDocumentData
+   * @returns SVGDocument
+   */
+  protected buildDocument(data: SVGDocumentData): SVGDocument {
+    return {
+      size: data.size,
+      nodes: data.nodes.map(ch => this.buildShape(ch)),
+    };
+  }
+
+  /**
+   * Build SVGShapeData into SVGShape
+   * @param data - SVGShapeData
+   * @returns SVGShape
+   */
+  protected buildShape({id, type, props, children}: SVGShapeData): SVGShape {
+    return {
+      id,
+      shape: new type({
+        children: children?.map(ch => this.buildShape(ch).shape),
+        ...this.processElementStyle(props),
+      }),
+    };
+  }
+
+  /**
+   * Parse SVG as SVGDocument
+   * @param svg - svg string to be parsed
+   * @returns - SVGDocument
+   */
+  protected parseSVG(svg: string): SVGDocument {
+    return this.buildDocument(SVG.parseSVGData(svg));
+  }
+
+  /**
+   * Create tweening list to tween node from initial node to final node.
+   * @param from - Initial node
+   * @param to - Final node
+   * @param time - time for tweening
+   * @param timing - timing for tweening
+   */
+  protected *generateTransformator(
```

Comment:
`Transformator` -> `Transformer`. There are a few other occurrences like this in the PR.